resources/assets/js/views/superadmin/components/TableList.vue

```diff
@@ -114,143 +560,504 @@
+      sortDirection = 'desc';
+
+    const queryString = window.location.search;
+
+    let urlParams = new URLSearchParams(queryString);
+    let orderType = ref('');
+    orderType.value = urlParams.get('orderBy') ?? '';
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    let range = '';
+
+    const sortingDirection = () => {
+      return sortParams.value.direction === 'asc' ? 'descending' : 'ascending';
+    };
+
+    const sortBy = async (order) => {
+      resetPagination.value = true;
+      sortParams.value.orderBy = order;
+      sortParams.value.direction =
+        sortParams.value.direction === 'desc' ? 'asc' : 'desc';
+
+      if (currentURL.includes('?')) {
+        query = urlParams.get('q') ?? '';
+        sortDirection = urlParams.get('direction') === 'desc' ? 'asc' : 'desc';
+
+        let startDate = urlParams.get('start_date') ?? false;
+        let endDate = urlParams.get('end_date') ?? false;
+
+        if (startDate && endDate) {
+          range = `&start_date=${startDate}&end_date=${endDate}`;
+        }
+      }
+
+      await fetchOrganisation(1);
+      resetPagination.value = false;
+    };
+
+    watch(
+      () => [
+        filter.country,
+        filter.completeness,
+        filter.registration_type,
+        filter.publisher_type,
+        filter.data_license,
+        filter.start_date,
+        filter.end_date,
+        filter.date_type,
+      ],
+      () => {
+        fetchOrganisation();
+      },
+      { deep: true }
+    );
+
+    const resetAllFilters = () => {
+      ignoreUpdates(() => {
+        filter.country = [];
+        filter.publisher_type = [];
+        filter.data_license = [];
+        filter.completeness = '';
+        filter.registration_type = '';
+        filter.start_date = '';
+        filter.end_date = '';
+        filter.date_type = 'created_at';
+        filter.selected_date_filter = '';
+      });
+    };
+
+    const isFilterApplied = computed(() => {
+      return (
+        filter.country.length +
+          filter.publisher_type.length +
+          filter.data_license.length !=
+          0 ||
+        filter.completeness !== '' ||
+        filter.registration_type !== '' ||
+        (filter.start_date !== '' && filter.end_date !== '')
+      );
+    });
+
+    const refreshStatusArrays = (orgData) => {
+      for (let orgDatum of orgData.data) {
+        registryApiKeyStatus[orgDatum.id] =
+          orgDatum?.settings?.publishing_info?.token_verification ?? false;
+        defaultValueStatus[orgDatum.id] = checkIfDefaultValuesAreValid(
+          orgDatum ? orgDatum.settings : false
+        );
+      }
+    };
+
+    const checkIfDefaultValuesAreValid = (settings) => {
+      if (settings) {
+        let defaultValues = settings.default_values;
+        let activityDefaultValues = settings.activity_default_values;
+        return !!(
+          (defaultValues?.default_currency ?? false) &&
```

Comment:
will require additional checks for default values now contain other values like collaboration type, default flow type, default aid type etc

Reply:
agreed.